tests/src/test/fdc3.broadcast.ts

```diff
@@ -822,36 +1119,74 @@ export default () =>
       }
     }
 
-    function resetChannelsAppContext() {
-      channelsAppContext.broadcastMultipleItems = false;
-      channelsAppContext.useAppChannel = false;
-      channelsAppContext.contextBroadcasts.contact = false;
-      channelsAppContext.reverseMethodCallOrder = false;
-      channelsAppContext.broadcastExecutionComplete = false;
-    }
-
-    const executionCompleteListener = (
+    const waitForContext = (
       contextType: string,
+      testId: string,
       channel?: Channel
     ) => {
       return new Promise<Context>(async (resolve) => {
-        if (channel === undefined) {
-          const listener = await window.fdc3.addContextListener(
-            contextType,
-            (context) => {
+        const handler = (context: AppControlContext) => {
+          if (testId) {
+            if (testId == context.testId) {
               resolve(context);
               listener.unsubscribe();
+            } else {
+              console.warn(
+                `Ignoring ${contextType} context due to mismatched testId (expected: ${testId}, got ${context.testId})`
+              );
             }
+          } else {
+            resolve(context);
+            listener.unsubscribe();
+          }
+        };
+        if (channel === undefined) {
+          const listener = await window.fdc3.addContextListener(
+            contextType,
+            handler
           );
         } else {
+          //App channels do not auto-broadcast current context when you start listening, so retrieve currnet context to avoid races
```

Comment:
```suggestion
          //App channels do not auto-broadcast current context when you start listening, so retrieve current context to avoid races
```